server/src-lib/Hasura/App.hs

```diff
@@ -213,7 +216,8 @@ runHGEServer ServeOptions{..} InitCtx{..} initTime = do
 
   authMode <- either (printErrExit . T.unpack) return authModeRes
 
-  HasuraApp app cacheRef cacheInitTime shutdownApp <-
+  let flushLogger = liftIO $ FL.flushLogStr $ _lcLoggerSet loggerCtx
+  HasuraApp app cacheRef cacheInitTime shutdownApp <- flip onException flushLogger $
```

Comment:
Please add a small comment that references issue #4772 so that it’s clear why this is needed! Otherwise this LGTM.